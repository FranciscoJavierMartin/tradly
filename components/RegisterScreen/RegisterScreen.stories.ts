import RegisterScreen from './RegisterScreen.vue';

export default {
  title: 'RegisterScreen',
  component: RegisterScreen,
  args: {},
  parameters: {
    layout: 'fullscreen',
  },
};

const Template = (args) => ({
  components: { RegisterScreen },
  setup() {
    return { args };
  },
  template: '<RegisterScreen v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};
