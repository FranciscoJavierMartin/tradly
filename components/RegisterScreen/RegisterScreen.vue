<template>
  <div class="c-register-screen">
    <h1>Welcome to tradly</h1>
    <h6>Signup to your account</h6>
    <form>
      <CustomInput type="text" placeholder="First Name" />
      <CustomInput type="text" placeholder="Last Name" />
      <CustomInput type="email" placeholder="Email" />
      <CustomInput type="password" placeholder="Password" />
      <CustomInput type="password" placeholder="Confirm Password" />
      <CustomButton text="Create" inverted />
    </form>
    <span class="c-register-screen__sign-in">
      Have an account? <a href="#">Sign in</a>
    </span>
  </div>
</template>

<script setup lang="ts">
import CustomButton from '../CustomButton/CustomButton.vue';
import CustomInput from '../CustomInput/CustomInput.vue';
const props = defineProps({});
</script>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-register-screen {
  background-color: $dark-green;
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  color: $white;

  h1 {
    padding-top: 40px;
    font-size: 1.5rem;
    font-weight: 500;
    line-height: 29px;
  }

  h6 {
    font-size: 16px;
    font-weight: 400;
    line-height: 19px;
  }

  form {
    display: flex;
    flex-direction: column;
    padding: 0px 20px;

    input {
      margin-bottom: 16px;
    }

    button {
      margin-top: 25px;
    }
  }

  &__sign-in {
    margin-top: 30px;
    font-weight: 400;
    font-size: 18px;
    line-height: 22px;
    padding-bottom: 30px;

    a {
      font-weight: 600;
      text-decoration: none;
      color: $white;
    }
  }
}
</style>
