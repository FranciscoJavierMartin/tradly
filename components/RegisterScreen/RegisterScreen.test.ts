import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/vue';
import RegisterScreen from './RegisterScreen.vue';

describe('RegisterScreen.vue', () => {
  test('should render', () => {
    const { container } = render(RegisterScreen);

    expect(
      screen.getByRole('heading', { name: /Welcome to tradly/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /Signup to your account/i })
    ).toBeInTheDocument();

    expect(screen.getByPlaceholderText('First Name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Last Name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Confirm Password')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Create/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /Sign in/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
