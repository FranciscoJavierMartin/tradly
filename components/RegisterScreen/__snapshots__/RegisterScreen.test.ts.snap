// Vitest Snapshot v1

exports[`RegisterScreen.vue > should render 1`] = `
<div
  class="c-register-screen"
>
  <h1>
    Welcome to tradly
  </h1>
  <h6>
    Signup to your account
  </h6>
  <form>
    <input
      class="c-custom-input"
      placeholder="First Name"
      type="text"
    />
    <input
      class="c-custom-input"
      placeholder="Last Name"
      type="text"
    />
    <input
      class="c-custom-input"
      placeholder="Email"
      type="email"
    />
    <input
      class="c-custom-input"
      placeholder="Password"
      type="password"
    />
    <input
      class="c-custom-input"
      placeholder="Confirm Password"
      type="password"
    />
    <button
      class="c-custom-button c-custom-button--inverted"
    >
      Create
    </button>
  </form>
  <span
    class="c-register-screen__sign-in"
  >
     Have an account? 
    <a
      href="#"
    >
      Sign in
    </a>
  </span>
</div>
`;
